openfisca_france/model/prestations/minima_sociaux/aah.py

```diff
@@ -319,8 +318,13 @@ def formula(individu, period):
         ressources = sum(
             [individu(ressource, three_previous_months, options = [ADD]) for ressource in ressources_a_inclure]
             )
+        
+        # On récupère le montant absolu des pensions alimentaires versées au cas où la valeur reçue est négative
+        pensions_alimentaires_versees = abs_(individu(
+            'pensions_alimentaires_versees_individu', three_previous_months, options = [ADD]))
 
-        return ressources * 4
+        # On soustrait le montant des pensions alimentaires versées à la base des ressources
+        return (ressources - pensions_alimentaires_versees) * 4
```

Comment:
On suppose ici qu'on n'aura jamais de pension alimentaire versée supérieure aux ressources de l'individu sachant que ce sont les revenus nets catégoriels de l'impôt qui sont pris en compte d'après [service-public](https://www.service-public.fr/particuliers/vosdroits/F12242) et [cette note DREES (pdf 2022)](https://drees.solidarites-sante.gouv.fr/sites/default/files/2022-09/Minima22%20Fiche%209%20-%20L’assiette%20des%20ressources%20et%20la%20période%20de%20référence%20des%20prestations.pdf). 

Néanmoins, probablement sur de plus rares cas, si l'individu a des ressources d'un autre pays non soumises à l'impôt en France, on pourrait obtenir une base de ressources négative (une sorte d'avantage en base de ressources) d'où cette proposition pour éviter ça :

```suggestion
        return max_(0, ressources - pensions_alimentaires_versees) * 4
```

Reply:
Avec cette solution, si l'individu n'a pas de ressources hors activité alors les pensions alimentaires versées ne sont pas prises en compte.
Je commit une solution avec les tests associés pour revues 😊👍